### PR TITLE
Elaborate on instructions for home assistant deployed using docker container

### PR DESCRIPTION
**Pull request by srwareham**
Elaborate on instructions for home assistant deployed using docker container

## Proposed change
Documentation update describing how weatherflow integration needs to be configured when home assistant is run from a docker container.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- This PR fixes or closes issue: fixes https://github.com/home-assistant/core/issues/101423

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **Documentation**
  - Updated instructions for deploying Home Assistant via a Docker container with specific networking configurations for proper UDP communication on port 50222.

### CODE REVIEW

File: source/_integrations/weatherflow.markdown [hunk 1]
```diff
@@ -60,4 +60,22 @@ Additionally the following diagnostic sensors are available:
...
+  --network=host \
+  ghcr.io/home-assistant/home-assistant:stable
+```
+
+or
+
+```
+docker run -d \
+   ...
+  -p 50222:50222/udp \
```
Comment (frenck) [inline source/_integrations/weatherflow.markdown:77]: This should make no difference, as Home Assistant must run on the host network.

Could you elaborate on this change?

../Frenck

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_integrations/weatherflow.markdown (hunk 1 above)
Comment (frenck) [reply source/_integrations/weatherflow.markdown:77]: Aah I see now, this is trying to add instruction for the non-hostwork case.

Unfortunately, the Home Assistant project does not support running without hostnetwork support. Therefore, this should be in our official documentation.

../Frenck

Comment (srwareham) [reply source/_integrations/weatherflow.markdown:77]: Understood it's preferred to keep the docs focused on the supported configuration. Would you be opposed to an addition that states:

"If you are deploying Home Assistant via a Docker container, make sure to run using host networking mode. You may see the error `Config flow could not be loaded: Unknown error` if Home Assistant is unable to receive `UDP` traffic on port `50222`.

I could be helpful to add some SEO discoverability for the specific issue given the error message is a generic one. Not sure if its practical to actually catch this error in the actual integrations code, preempting the generic error.

Comment (frenck) [reply source/_integrations/weatherflow.markdown:77]: We do not support that use case. The correct solution is to install Home Assistant as documented.

**Pull request closed without merging**